src/main/java/com/minecolonies/coremod/colony/workorders/WorkOrderBuild.java

```diff
@@ -99,7 +100,23 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
             upgradeName = compound.getString(TAG_UPGRADE_NAME);
         }
         cleared = compound.getBoolean(TAG_IS_CLEARED);
+        md5 = compound.getString(TAG_SCHEMATIC_MD5);
         structureName = compound.getString(TAG_SCHEMATIC_NAME);
+        if (Structures.getMD5(structureName).isEmpty())
+        {
+            if (!Structures.getMD5(getStructurePrefix() + '/' + structureName).isEmpty())
```

Comment:
Again see comment regarding line 325 of the AbstractBuilding class:

It might be a good idea to add methods to the Structures class to check if it has an MD5 or a Structure Name.
This would make these arbitrary checks against isEmpty() not needed. Improve the readability and allow for more flexibility in the future, especially when creating an API for this kind of stuff, arbitrary checks should be avoided if they can simple be wrapped inside a proper method.